pkg/vmconfig/bootoptions/bootoptions_reconciler.go

```diff
@@ -119,3 +131,135 @@ func (r reconciler) Reconcile(
 
 	return nil
 }
+
+func reconcileBootOrder(
+	ctx context.Context,
+	k8sClient ctrlclient.Client,
+	vm *vmopv1.VirtualMachine,
+	moVM mo.VirtualMachine,
+	vmBootOptions *vmopv1.VirtualMachineBootOptions,
+	ci *vimtypes.VirtualMachineConfigInfo) ([]vimtypes.BaseVirtualMachineBootOptionsBootableDevice, error) {
+
+	devices := object.VirtualDeviceList(ci.Hardware.Device)
```

Comment:
Change this to a non-pointer type since you're just reading the `ci.Hardware.Device` in this function?